### PR TITLE
Update readme with instructions for bam, minor fixes

### PR DESCRIPTION
**Pull request by Dune-jr**
Some say bam should be phased out and replaced by CMake. I had people think bam was deprecated because of the lack of instructions in the readme. The ambiguity should be cleared up if bam support is to be continued.

Note I left CMake first in all sections.

### CODE REVIEW

**Comment by heinrich5991** [comment]
>[4:09 PM] heinrich5991: not sure I agree that it is beneficial to put into the readme though
>[4:09 PM] heinrich5991: for newcomers, the "+ used it before" goes away
>[4:10 PM] Dune: Honestly, I wrote those changes long ago, and ended up not PR-ing because I thought that would recreate unnecessary friction and that the initial readme was written without intents of pushing bam away
>[4:11 PM] Dune: Newcomers will probably try the first sections anyway
>[4:12 PM] heinrich5991: probably. a little complexity is added to the readme due to that, though

"+ used it before" was from the listing of advantages of bam. The advantage stated was that bam is the build tool Teeworlds has used so the Teeworlds developers are used to it.

**Comment by Learath2** [comment]
Seems the dependencies for bam and cmake are very similar, maybe we could merge that section.

**Comment by Dune-jr** [comment]
@Learath2 Here, an old draft with an attempt to do that: https://github.com/Dune-jr/teeworlds/blob/task-readme-bam/readme.md but I believe it is harder to follow

**Comment by Learath2** [comment]
I agree this seems more readable, you could however move the "Downloading repository" part out of the bam part and make it precede all builds.

File: readme.md [hunk 1]
```diff
@@ -23,22 +23,25 @@ the game, including new versions, custom maps and much more.
 
 Originally written by Magnus Auvinen.
 
+---
 
-Building on Linux or macOS
+Teeworlds supports two build systems: CMake (widely used) and bam (fast build times).
```
Comment (Learath2) [inline readme.md:28]: This is still not true for me, I can only observe bam being faster if you forget to configure your `make`.
https://gist.github.com/Learath2/e62c57066859122b1f584c4d8ebaf157

bam does indeed have an edge while configuring, but again it configures much less so it is expected.

Comment (Dune-jr) [reply readme.md:28]: That is the first time I hear about bam being slower than `make`. In all the experiments I did (with `-j`, of course) make was much slower. Ninja was slower on my usual machine (18s vs 13s), and similar on a high performance 12-core machine (3s vs 3s).

### Build
make -j4  65.99s user 4.10s system 368% cpu 19.038 total

ninja  65.54s user 3.26s system 378% cpu 18.189 total

bam conf=release  49.35s user 2.99s system 390% cpu 13.405 total

bam  31.62s user 3.78s system 382% cpu 9.245 total

### Config
cmake ..  2.13s user 0.78s system 101% cpu 2.875 total

bam config  0.10s user 0.00s system 100% cpu 0.104 total

**Comment by Dune-jr** [comment]
> I agree this seems more readable, you could however move the "Downloading repository" part out of the bam part and make it precede all builds.

That is true, we should do that.

File: readme.md (hunk 1 above)
Comment (Dune-jr) [reply readme.md:28]: Office machine (8 core)

### Build
make -j8
real	0m9.049s
user	0m48.333s
sys	0m2.845s

ninja
real	0m8.676s
user	0m47.556s
sys	0m2.362s

bam conf=release
real	0m6.880s
user	0m43.153s
sys	0m2.496s

### Config
cmake ..
real	0m2.004s
user	0m1.441s
sys	0m0.568s

bam config
real	0m0.241s
user	0m0.169s
sys	0m0.193s

Comment (heinrich5991) [reply readme.md:28]: Here's my low-end machine (Arch Linux, Intel(R) Celeron(R) CPU 1007U @ 1.50GHz):

Build
=====

ninja
Executed in   44.17 secs   fish           external 
   usr time   68.03 secs    0.23 millis   68.03 secs 
   sys time    7.99 secs    3.65 millis    7.99 secs 

bam
Executed in   46.82 secs   fish           external 
   usr time   70.65 secs    0.00 millis   70.65 secs 
   sys time   10.35 secs    4.32 millis   10.34 secs 


Config
======

cmake .. -DDEV=ON -GNinja
Executed in    6.83 secs   fish           external 
   usr time    4.87 secs  685.36 millis    4.19 secs 
   sys time    2.35 secs   24.05 millis    2.33 secs 

bam config
Executed in    1.02 secs   fish           external 
   usr time  982.01 millis    0.00 millis  982.01 millis 
   sys time  890.11 millis    5.06 millis  885.04 millis

Comment (heinrich5991) [reply readme.md:28]: What OS are you building on, @Dune-jr?

Comment (Dune-jr) [reply readme.md:28]: @heinrich5991 Linux Mint 19.3, cmake 3.10.2, GNU Make 4.1, ninja 1.8.2

Comment (Learath2) [reply readme.md:28]: macOS 10.15, cmake 3.17.2, GNU Make 3.81, ninja 1.10.0 is what I ran this test on

Comment (heinrich5991) [reply readme.md:28]: cmake 3.17.1, GNU Make 4.3, ninja 1.10.0, bam 0.5.1 using Lua 5.3.3 for me.

File: readme.md [hunk 2]
```diff
@@ -110,9 +165,33 @@ arrow. Wait for the compilation to finish.
...
+Download and unzip [Teeworlds stable sources](https://github.com/teeworlds/teeworlds/releases) or [Teeworlds latest sources](https://github.com/teeworlds/teeworlds/archive/master.zip).
+
+Download and unzip [bam](https://github.com/matricks/bam/archive/v0.5.1.tar.gz) to `teeworlds-version\bam`.
+
+Run the `x64 Native Tools Command Prompt` (or `x86` for 32-bit) from the start menu.
 
-Building on Windows with MinGW
-==============================
+    # Navigate to the Teeworlds source directory
+    cd ...\teeworlds-version
```
Comment (AssassinTee) [inline readme.md:180]: 3 dots?

**Pull request merged**